include/swift/AST/Expr.h

```diff
@@ -4787,10 +4788,10 @@ class KeyPathExpr : public Expr {
     } Decl;
     
     
-    llvm::PointerIntPair<Expr *, 3, Kind> SubscriptIndexExprAndKind;
+    llvm::PointerIntPair<Expr *, 3, unsigned> SubscriptIndexExprAndKindLo;
     ArrayRef<Identifier> SubscriptLabels;
     ArrayRef<ProtocolConformanceRef> SubscriptHashableConformances;
-    Type ComponentType;
+    llvm::PointerIntPair<Type, 1, unsigned> ComponentTypeAndKindHi;
```

Comment:
Component is already 9 words, so rather than cramming things in even tighter here it may be worth just making a separate field. (You could probably also shrink things down by taking the counts out of SubscriptLabels and SubscriptHashableConformances, getting down to 8 words and still having space left over for the kind.)